client/src/components/AuthForm.js

```diff
@@ -0,0 +1,171 @@
+import React, { useState, useRef } from "react";
+import PropTypes from "prop-types";
+import TextField from "@material-ui/core/TextField";
+import { makeStyles } from "@material-ui/core/styles";
+import { Button } from "@material-ui/core";
+import { useAuth } from "../utils/AuthProvider";
+import _ from "lodash";
+const useStyles = makeStyles(theme => ({
+    root: {},
+    container: {},
+    textField: {},
+
+    displayName: {}
+}));
+
+const AuthForm = props => {
+    const { name, displayName } = props;
+    const classes = useStyles();
+    const [authForm, setAuthForm] = useState({
+        email: "",
+        password: "",
+        name: "",
+        confirmPassword: ""
+    });
```

Comment:
I feel like instead of having one state variable value here, we should have separate state values for each of these fields? What do you think?

Reply:
sure, I've just made those changes